File: PathoFroh/src/main/java/com/patho/main/action/dialog/miscellaneous/ConfirmDialog.java
package com.patho.main.action.dialog.miscellaneous;

import org.springframework.beans.factory.annotation.Configurable;

import com.patho.main.action.dialog.AbstractDialog;
import com.patho.main.common.Dialog;
import com.patho.main.util.dialogReturn.DialogReturnEvent;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Configurable
public class ConfirmDialog extends AbstractDialog {

	private String headline;
	private String text;

	public ConfirmDialog initAndPrepareBean() {
		return initAndPrepareBean("", "");
	}

	public ConfirmDialog initAndPrepareBean(String healine, String text) {
		if (initBean(healine, text))
			prepareDialog();
		return this;
	}

	public boolean initBean(String healine, String text) {
		setHeadline(healine.equals("") ? "" : resourceBundle.get(headline));
		setText(text.equals("") ? "" : resourceBundle.get(text));
		return super.initBean(Dialog.CONFIRM_CHANGE);
	}

	public ConfirmDialog header(String text) {
		return header(text, "");
	}

	public ConfirmDialog header(String text, Object... args) {
		setHeadline(resourceBundle.get(text, args));
		return this;
	}

	public ConfirmDialog ctext(String text) {
		return ctext(text, "");
	}

	public ConfirmDialog ctext(String text, Object... args) {
		setText(resourceBundle.get(text, args));
		return this;
	}

	public void confirmAndHide() {
		hideDialog(new ConfirmEvent(true));
	}

	@Getter
	@Setter
	@AllArgsConstructor
	public static class ConfirmEvent implements DialogReturnEvent {
		private boolean confirm;
	}

}

File: src/main/kotlin/com/patho/main/dialog/miscellaneous/ConfirmDialog.kt
package com.patho.main.dialog.miscellaneous

import com.patho.main.common.Dialog
import com.patho.main.dialog.AbstractDialog_
import com.patho.main.util.dialog.event.ConfirmEvent
import org.springframework.stereotype.Component

/**
 * Generic confirm dialog
 */
@Component()
open class ConfirmDialog : AbstractDialog_(Dialog.CONFIRM_CHANGE) {

    lateinit var headline: String

    lateinit var text: String

    open fun initAndPrepareBean(headline: String = "", body: String = ""): ConfirmDialog {
        if (initBean(headline, body))
            prepareDialog()
        return this
    }

    open fun initBean(headline: String = "", body: String = ""): Boolean {
        this.headline = if (headline.isEmpty()) "" else resourceBundle[headline]
        this.text = if (body.isEmpty()) "" else resourceBundle[body]
        return super.initBean()
    }

    open fun header(text: String): ConfirmDialog {
        return header(text, "")
    }

    open fun header(text: String, vararg args: Any): ConfirmDialog {
        headline = resourceBundle[text, args]
        return this
    }

    open fun ctext(text: String): ConfirmDialog {
        return ctext(text, "")
    }

    open fun ctext(text: String, vararg args: Any): ConfirmDialog {
        this.text = resourceBundle[text, args.map { it.toString() }]
        return this
    }

    open fun confirmAndHide() {
        return hideDialog(ConfirmEvent(true))
    }

    open fun declineAndHide() {
        return hideDialog(ConfirmEvent(false))
    }
}
